backtrace-library/src/main/java/backtraceio/library/common/ApplicationMetadataCache.java

```diff
@@ -6,44 +6,68 @@
 
 import backtraceio.library.logger.BacktraceLogger;
 
-public class ApplicationHelper {
-    private static final transient String LOG_TAG = ApplicationHelper.class.getSimpleName();
+public class ApplicationMetadataCache {
+
+    private static final transient String LOG_TAG = ApplicationMetadataCache.class.getSimpleName();
+
+    private static ApplicationMetadataCache instance;
+
     /**
      * Cached application name
      */
-    private static String applicationName;
+    private String applicationName;
 
     /**
      * Cached application version
      */
-    private static String applicationVersion;
+    private String applicationVersion;
 
     /**
      * Cached package name
      */
-    private static String packageName;
+    private String packageName;
+
+    /**
+     * Returns current application cache. This instance is a singleton since we can only operate
+     * in a single application scope.
+     *
+     * @param context Application context
+     * @return Application metadata cache
+     */
+    public static ApplicationMetadataCache getInstance(Context context) {
+        if (instance == null) {
+            instance = new ApplicationMetadataCache(context);
+        }
+        return instance;
+    }
```

Comment:
subsequent calls with different contexts will return the same instance. Is that something we want?

Reply:
We expect to have a single app context

Reply:
then maybe throw an error if the context is different?

Reply:
There is no way you can have two app context. I'm not even sure if this is even possible. The description makes it clear. I don't think it's a good thing. I prefer not to include this